packages/oauth/src/callback-options.ts

```diff
@@ -4,6 +4,39 @@ import { InstallURLOptions } from './install-url-options';
 import { Installation, OrgInstallation } from './installation';
 
 export interface CallbackOptions {
+
+  /**
+   * An additional logic to run right before executing the Slack app installation with the given code parameter.
+   *
+   * When this method returns false, the InstallProvider skips the installation.
+   * You can set false when the visiting user is not eligible for proceeding the Slack app installation flow.
+   *
+   * Also, when returning false, this method is responsible to call response#end() method
+   * to build complete HTTP response for end-users.
+   */
+  beforeInstallation?: (
+    options: InstallURLOptions,
+    callbackReq: IncomingMessage,
+    callbackRes: ServerResponse,
+  ) => Promise<boolean>;
+
+  /**
+   * An additional logic to run right after executing the Slack app installation with the given code parameter.
+   *
+   * When this method returns false, the InstallProvider skips storing the installation in database.
+   * You can set false when your app needs to cancel the installation (you can call auth.revoke API method for it)
```

Comment:
Just for my own understanding: what could be a potential situation for returning `true` in `beforeInstallation` but `false` in `afterInstallation`? Maybe some kind of resource authorization specific in the third party service? I.e. after logging in to the third party service, it turns out the user does not have access to whatever resource they require for use of the app? This comment is only for my own learning 🙇

Reply:
Good to know, thank you!